PostHog/PostHogSDK.swift

```diff
@@ -301,18 +301,23 @@ let maxRetryDelay = 30.0
             props["$process_person_profile"] = hasPersonProcessing()
         }
 
+        let sdkInfo = context?.sdkInfo()
+        if sdkInfo != nil {
+            props = props.merging(sdkInfo ?? [:]) { current, _ in current }
+        }
+
         if let sessionId = PostHogSessionManager.shared.getSessionId() {
             props["$session_id"] = sessionId
-            // Session replay requires $window_id, so we set as the same as $session_id.
+            // only Session replay requires $window_id, so we set as the same as $session_id.
             // the backend might fallback to $session_id if $window_id is not present next.
             #if os(iOS)
-                if config.sessionReplay {
+                if !appendSharedProps, config.sessionReplay {
```

Comment:
@pauldambra just to confirm
events only need `$session_id` and not `$window_id`
recordings need both, right?

Reply:
we put $window_id on all events from JS.

I can't believe that we put $window_id on all events from e.g. the Go SDK so it _should_ be safe to not add $window_id where it's not meaningful

I think either in capture or at query time we'd be safe to say $window_id == $session_id if not present